x-pack/plugins/index_management/public/application/sections/home/data_stream_list/data_stream_detail_panel/data_stream_detail_panel.tsx

```diff
@@ -273,15 +298,58 @@ export const DataStreamDetailPanel: React.FunctionComponent<Props> = ({
         toolTip: i18n.translate('xpack.idxMgmt.dataStreamDetailPanel.dataRetentionToolTip', {
           defaultMessage: 'The amount of time to retain the data in the data stream.',
         }),
-        content: getLifecycleValue(lifecycle),
+        content: (
+          <ConditionalWrap
+            condition={isDataStreamFullyManagedByILM(dataStream)}
+            wrap={(children) => <s>{children}</s>}
+          >
+            <>{getLifecycleValue(lifecycle)}</>
+          </ConditionalWrap>
+        ),
         dataTestSubj: 'dataRetentionDetail',
       },
     ];
 
     const managementDetails = getManagementDetails();
     const details = [...defaultDetails, ...managementDetails];
 
-    content = <DetailsList details={details} />;
+    content = (
+      <>
+        {isDataStreamFullyManagedByILM(dataStream) && (
+          <>
+            <EuiCallOut
+              title={i18n.translate(
+                'xpack.idxMgmt.dataStreamsDetailsPanel.editDataRetentionModal.fullyManagedByILMTitle',
+                { defaultMessage: 'This data stream is fully managed by ILM' }
+              )}
+              iconType="pin"
+              data-test-subj="dsIsFullyManagedByILM"
+            >
+              <p>
+                <FormattedMessage
+                  id="xpack.idxMgmt.dataStreamsDetailsPanel.editDataRetentionModal.fullyManagedByILMDescription"
+                  defaultMessage="Editing the data retention for this data stream is only possible through its {link}."
```

Comment:
```suggestion
                  defaultMessage="To edit data retention for this data stream, you must edit its associated {link}."
```
Minor styling edit for a more active voice.